xcvm/xcvm-instructions/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use alloc::vec::Vec;
+
+extern crate alloc;
+
+pub trait XCVMProtocol {
+	fn serialize(&self, network: &XCVMNetwork) -> Vec<u8>;
+}
+
+#[derive(Copy, Clone)]
+pub struct Stableswap<Assets>(Assets, Assets);
+
+impl<Assets> XCVMProtocol for Stableswap<Assets> {
+	fn serialize(&self, network: &XCVMNetwork) -> Vec<u8> {
+		match network {
+			XCVMNetwork::Substrate(XCVMSubstrateChain::Picasso) => todo!("hardcoded"),
+			XCVMNetwork::Cosmos(XCVMCosmChain::Terra) => todo!("hardcoded"),
+		}
+	}
+}
+
+#[derive(Copy, Clone)]
+pub enum XCVMSubstrateChain {
+	Picasso,
+}
+
+#[derive(Copy, Clone)]
+pub enum XCVMCosmChain {
+	Terra,
+}
+
+#[derive(Copy, Clone)]
+pub enum XCVMNetwork {
+	Substrate(XCVMSubstrateChain),
+	Cosmos(XCVMCosmChain),
+}
```

Comment:
I would simplify this a bit, perhaps:

```
pub enum Network {
    Picasso,
    Osmosis,
    Custom(u32)
}
```

Where we can doing storage accesses for the well know networks and hardcode routes (Picasso <> KSM), but for future use, add more networks with extrinsics.

Reply:
Resolved by changing to

```
#[repr(transparent)]
pub struct XCVMNetwork(u8);
```